cmd/fetch-failed-logs.rb

```
@@ -0,0 +1,110 @@
+    end
+
+    # Print only interesting lines
+    puts content[brew_index..failed_index]
+  end
+
+  def fetch_failed_logs
+    fetch_failed_logs_args.parse
+
+    raise FormulaUnspecifiedError if Homebrew.args.named.empty?
+
+    formula = Homebrew.args.resolved_formulae.first
+    workflow = args.dispatched? ? "dispatch-build-bottle.yml" : "build-bottles.yml"
+    repo = "Homebrew/linuxbrew-core"
+
+    # First get latest workflow runs
+    url = "https://api.github.com/repos/#{repo}/actions/workflows/#{workflow}/runs?status=failure"
+    response = GitHub.open_api(url, request_method: :GET, scopes: ["repo"])
+    workflow_runs = response["workflow_runs"]
+
+    # Then iterate over them and find the matching one...
+    workflow_run = workflow_runs.find do |run|
+      # If the workflow run was triggered by a repository dispatch event, then
+      # check if any step name in all its jobs is equal to formula
+      if run["event"] == "repository_dispatch"
+        url = run["jobs_url"]
+        response = GitHub.open_api(url, request_method: :GET, scopes: ["repo"])
+        jobs = response["jobs"]
+        jobs.find do |job|
+          steps = job["steps"]
+          steps.find do |step|
+            step["name"].match(formula.name)
+          end
+        end
+      # If the workflow run was triggered by a pull request event, then
+      # fetch the head commit, determine which file changed and
+      # check if equal to formula
+      elsif run["event"] == "pull_request"
+        url = "https://api.github.com/repos/#{repo}/commits/#{run["head_sha"]}"
+        response = GitHub.open_api(url, request_method: :GET, scopes: ["repo"])
+        commit_files = response["files"].map { |f| f["filename"] }
+        commit_files.find do |file|
+          file[%r{Formula/(.+)\.rb}, 1] == formula
+        end
+        # Not supported event
+      end
+    end
+
+    unless workflow_run
+      opoo "No workflow run matching the criteria was found"
+      return
```

`odie` ?

---

`opoo` + `return` is fine too if you don't want to return an error when there is no matching workflow run.

---

Suggestion applied, thanks!